chia/wallet/wallet_puzzle_store.py

```diff
@@ -142,11 +142,11 @@ async def get_derivation_record_for_puzzle_hash(self, puzzle_hash: str) -> Optio
 
         if row is not None and row[0] is not None:
             return DerivationRecord(
-                row[0],
-                bytes.fromhex(row[2]),
+                uint32(row[0]),
+                bytes32.fromhex(row[2]),
```

Comment:
we really store binary data in hex form in the database?

Reply:
Yes, we currently do. See `add_derivation_paths()` in the same file.